docs/_docs/03-ug-mutable-dataset.md

```diff
@@ -0,0 +1,38 @@
+---
+title: "Mutable dataset"
+permalink: /docs/ug-mutable-dataset/
+excerpt: "Mutable dataset guide"
+last_modified_at: 2020-10-14
+toc: false
+classes: wide
+---
+
+This guide helps you to utilize indexes on mutable dataset.
+
+## What does mutable mean?
+Hyperspace can build indexes based on the source files in the given logical plan.
```

Comment:
Can we simplify this part? Something like: "Hyperspace enables users to build indexes on their data"? 
I guess talking about source files in a logical plan could be a bit confusing for an ordinary user who is not necessarily familiar with our code base.